src/pineko/fonll.py

```diff
@@ -0,0 +1,274 @@
+"""Module to manage FONLL predictions."""
+
+import copy
+import json
+import tempfile
+from pathlib import Path
+
+import numpy as np
+import pineappl
+import rich
+import yaml
+
+from . import configs
+
+
+class FONLLInfo:
+    """Class containing all the information for FONLL predictions."""
+
+    def __init__(
+        self, ffns3, ffn03, ffns4, ffns4til, ffns4bar, ffn04, ffns5, ffns5til, ffns5bar
+    ) -> None:
+        """Initialize fonll info."""
+        self.ffns3 = ffns3
+        self.ffn03 = ffn03
+        self.ffns4 = ffns4
+        self.ffns4til = ffns4til
+        self.ffns4bar = ffns4bar
+        self.ffn04 = ffn04
+        self.ffns5 = ffns5
+        self.ffns5til = ffns5til
+        self.ffns5bar = ffns5bar
+
+    @property
+    def fk_paths(self):
+        """Returns the list of the fk paths needed to produce FONLL predictions."""
+        paths = {
+            "ffns3": self.ffns3,
+            "ffn03": self.ffn03,
+            "ffns4": self.ffns4,
+            "ffns4til": self.ffns4til,
+            "ffns4bar": self.ffns4bar,
+            "ffn04": self.ffn04,
+            "ffns5": self.ffns5,
+            "ffns5til": self.ffns5til,
+            "ffns5bar": self.ffns5bar,
+        }
+        return {p: Path(paths[p]) for p in paths if paths[p] is not None}
+
+    @property
+    def fks(self):
+        """Returns the pineappl.Grid objects reading the paths provided by self.fk_paths."""
+        return {fk: pineappl.grid.Grid.read(self.fk_paths[fk]) for fk in self.fk_paths}
+
+    @property
+    def dataset_name(self):
+        """Return the name of the dataset given by the fktables name, if all the fktables have the same name."""
+        if len({self.fk_paths[p].name for p in self.fk_paths}) == 1:
+            return self.fk_paths[list(self.fk_paths)[0]].name
+        else:
+            raise ValueError("not all fktables have the same name")
+
+    @property
+    def theorycard_no_fns_pto(self):
+        """Return the common theory info between the different FONLL FKs."""
+        theorycards = [json.loads(self.fks[p].key_values()["theory"]) for p in self.fks]
+        # Only these should differ
+        for card in theorycards:
+            del card["FNS"]
+            del card["PTO"]
+            del card["NfFF"]
+            del card["ID"]
+            del card["FONLLParts"]
+            del card["Comments"]
+        if len(theorycards) > 1 and not all(
+            [theorycards[0] == card in theorycards[1:]]
+        ):
+            raise ValueError("theorycards not the same")
+        return theorycards[0]
+
+    @property
+    def Q2grid(self):
+        """Return the Q2grid of the fktables given by self.fks ."""
+        aa = json.loads(self.fks[list(self.fks)[0]].key_values()["runcard"])[
+            "observables"
+        ]
+        bb = list(aa.values())[
+            0
+        ]  # there is only a single obseravble because it's a dis fktable
+        cc = np.array([i["Q2"] for i in bb])
+        return cc
```

Comment:
```suggestion
        return np.array([i["Q2"] for i in bb])
```

To save one more random name